src/core/qgscoordinatetransform.cpp

```diff
@@ -648,12 +648,24 @@ void QgsCoordinateTransform::transformCoords( int numPoints, double *x, double *
 
   int projResult = 0;
 #if PROJ_VERSION_MAJOR>=6
+  const bool sourceAxisOrderSwapped = d->mSourceAxisOrderSwapped;
+  const bool destinationAxisOrderSwapped = d->mDestAxisOrderSwapped;
+
   proj_trans_generic( projData, direction == ForwardTransform ? PJ_FWD : PJ_INV,
-                      x, sizeof( double ), numPoints,
-                      y, sizeof( double ), numPoints,
+                      !sourceAxisOrderSwapped ? x : y, sizeof( double ), numPoints,
+                      !sourceAxisOrderSwapped ? y : x, sizeof( double ), numPoints,
                       z, sizeof( double ), numPoints,
                       nullptr, sizeof( double ), 0 );
   projResult = proj_errno( projData );
+  if ( projResult == 0 && destinationAxisOrderSwapped )
+  {
+    size_t size = sizeof( double ) * numPoints;
```

Comment:
Would this work (didn't test)?

`std::swap(std::begin( x ), std::begin( x ) + numPoints, std::begin( y ) );`

Reply:
Can't get that to build, sorry -- I think it would require changing the signature from the double * arguments to vectors. But this branch is conditional and is used for proj v6.0 only -- I understand that proj 6.1 will become the recommended version quickly, and use of .0 will be discouraged. So I'm not too concerned by this.

Reply:
ok, forget about it, was just a random idea